spark/src/main/scala/org/apache/comet/rules/CometScanRule.scala

```diff
@@ -96,6 +96,13 @@ case class CometScanRule(session: SparkSession) extends Rule[SparkPlan] {
           return withInfo(scanExec, fallbackReasons.mkString(", "))
         }
 
+        if (scanImpl == CometConf.SCAN_NATIVE_DATAFUSION && scanExec.bucketedScan) {
+          // https://github.com/apache/datafusion-comet/issues/1719
+          fallbackReasons +=
+            "Full native scan disabled because bucketed scan is not supported"
+          return withInfo(scanExec, fallbackReasons.mkString(", "))
```

Comment:
nit: you can call the version that takes a set rather than building a string here. This will need an import to be added.

```suggestion
          return withInfos(scanExec, fallbackReasons.toSet)
```